llvm/lib/Transforms/InstCombine/InstCombineLoadStoreAlloca.cpp

```diff
@@ -270,16 +270,54 @@ class PointerReplacer {
     unsigned ToAS = ASC->getDestAddressSpace();
     return (FromAS == ToAS) || IC.isValidAddrSpaceCast(FromAS, ToAS);
   }
+  bool foundASC(const Value *Op) const;
+  bool hasConflictingAS(const SelectInst *I) const;
 
   SmallPtrSet<Instruction *, 32> ValuesToRevisit;
+  SmallPtrSet<SelectInst *, 32> ValuesToRevisitAS;
   SmallSetVector<Instruction *, 4> Worklist;
   MapVector<Value *, Value *> WorkMap;
   InstCombinerImpl &IC;
   Instruction &Root;
   unsigned FromAS;
+  bool HasASC = false;
 };
 } // end anonymous namespace
 
+/// Return true iff valid addrspacecast is found on
+/// path from end of branch to root of tree.
+bool PointerReplacer::foundASC(const Value *Op) const {
+  const Instruction *TI;
```

Comment:
Can you just check if the access address space matches the root address space?

Reply:
Hm, will try this out. It appears to be, so then HasASC is moreso unnecessary since hasConflictingAS will be constant. I'll take a look again. Thank you.